sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -165,6 +166,7 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
       RemoveLiteralFromGroupExpressions,
       RemoveRepetitionFromGroupExpressions) :: Nil ++
     operatorOptimizationBatch) :+
+    // Join reorder should be FixedPoint(1) since cost can change anyway for AQP.
```

Comment:
Based on the SPARK-28530 description, how about a statement below;
```
Since join casts in AQP can change between multiple runs, there is no reason that we have an idempotence enforcement on this batch. We thus make it FixedPoint(1) instead of Once.
```

Reply:
@maropu join costs ? :-)

Reply:
!!!! my bad... thx, your comment ;) @dilipbiswal